modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelFactory.java

```diff
@@ -0,0 +1,453 @@
+
+    userSettings.ifPresent(mavenConfigurationBuilder::userSettingsLocation);
+
+    settingsSecurity.ifPresent(mavenConfigurationBuilder::settingsSecurityLocation);
+
+    return mavenConfigurationBuilder.build();
+  }
+
+  @Override
+  public DeployableProjectModel<M> createDeployableProjectModel() {
+    File pom = getPomFromFolder(projectFolder);
+    Model pomModel = getPomModelFromFile(pom);
+
+    ApplicationGAVModel appGAVModel =
+        new ApplicationGAVModel(pomModel.getGroupId(), pomModel.getArtifactId(), pomModel.getVersion());
+    ArtifactCoordinates artifactCoordinates = getApplicationArtifactCoordinates(pomModel, appGAVModel);
+
+    // Resolve dependencies
+    AetherMavenClient aetherMavenClient = new AetherMavenClient(mavenConfiguration);
+    DeployableDependencyResolver deployableDependencyResolver = new DeployableDependencyResolver(aetherMavenClient);
+
+    List<BundleDependency> dependencies = deployableDependencyResolver.resolveApplicationDependencies(pom, false, empty());
+    List<String> activeProfiles = mavenConfiguration.getActiveProfiles().orElse(emptyList());
+    List<Artifact> deployableDependencies =
+        updateArtifactsSharedState(dependencies, updatePackagesResources(toApplicationModelArtifacts(dependencies)),
+                                   pomModel, activeProfiles);
+
+    Map<ArtifactCoordinates, List<Artifact>> pluginsDependencies =
+        new DeployablePluginsDependenciesResolver().resolve(dependencies);
+
+    // Parse additional plugin dependencies
+    List<org.mule.runtime.module.artifact.activation.internal.plugin.Plugin> initialAdditionalPluginDependencies =
+        findArtifactPackagerPlugin(pomModel, activeProfiles)
+            .map(this::getAdditionalPluginDependencies).orElse(emptyList());
+
+    AdditionalPluginDependenciesResolver additionalPluginDependenciesResolver =
+        new AdditionalPluginDependenciesResolver(aetherMavenClient,
+                                                 initialAdditionalPluginDependencies,
+                                                 new File("temp"));
+
+    List<Plugin> additionalPluginDependencies = toPluginDependencies(additionalPluginDependenciesResolver
+        .resolveDependencies(dependencies, pluginsDependencies));
+
+    // Prepare bundle dependencies expected by the class loader configuration
+    List<org.mule.runtime.module.artifact.api.descriptor.BundleDependency> trueAppBundleDependencies =
+        deployableDependencies.stream()
+            .map(artifact -> createBundleDependencyFromPackagerDependency(uri -> uri).apply(artifact)).collect(toList());
+
+    // Get exported resources and packages
+    try {
+      getExportedPackagesAndResources();
+    } catch (IOException e) {
+      throw new ArtifactActivationException(createStaticMessage("Couldn't search exported packages and resources"), e);
+    }
+
+    // TODO: validate dependencies checking for incompatibilities
```

Comment:
this is not maven specific, this should be done with the generated object

Reply:
Moved.